common/lib/src/realm_types.dart

```diff
@@ -79,6 +79,20 @@ class RealmStateError extends StateError implements RealmError {
   RealmStateError(String message) : super(message);
 }
 
+/// An error type that describes a session-level error condition.
+/// /// {@category Sync}
+class SessionError extends SyncError {
+  /// If true the received error is fatal.
+  final bool isFatal;
+
+  SessionError(
+    String message,
+    this.isFatal, {
```

Comment:
isFatal could have a default value to false, but category should be required imo.

Reply:
Yes, sure.